spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+import neo
+
+from ..recordingextractor import RecordingExtractor
+from ..sortingextractor import SortingExtractor
+
+
+class _NeoBaseExtractor:
+    NeoRawIOClass = None
+
+    installed = True
+    is_writable = False
+    extractor_gui_params = []
+    
+    def __init__(self, **kargs):
+        self.neo_reader = self.NeoRawIOClass(**kkargs)
+        self.neo_reader.parse_header()
+        
+        # TODO propose a meachanisim to select the appropriate segment
+        # in case there are several
+        num_block = self.neo_reader.block_count()
+        assert num_block == 1, 'This file is multi block spikeextractors support only one segment'
+        num_seg = self.neo_reader.segment_count()
+        assert num_seg == 1, 'This file is multi segment spikeextractors support only one segment'
+        
+        # select first block first segment
+        self.block_index = 0
+        self.seg_index = 0
+    
+
+class NeoBaseRecordingExtractor(RecordingExtractor, _NeoBaseExtractor):
+    
+    def __init__(self, **kargs):
+        _NeoBaseExtractor.__init__(self, **kargs)
+        
+        
+        # TODO propose a meachanisim to select the appropriate channel groups
+        # in neo one channel group have the same dtype/sampling_rate/group_id
+        num_chan_group = self.neo_reader.get_group_channel_indexes()
+        assert num_chan_group == 1, 'This file have several channel groups spikeextractors support only one groups'
+        
+        
+        # spikeextractor for units to be uV implicitly
+        # check that units are V, mV or uV
+        # otherwise raise error
+        units = self.neo_reader.header['signal_channels']['units']
+        assert np.all(np.isin(units, ['V', 'mV', 'uV'])), 'Signal units no Volt compatible'
+        self.additional_gain = np.ones(units.size, dtype='float')
+        self.additional_gain[units == 'V'] = 1e6
+        self.additional_gain[units == 'mV'] =  1e3
+        self.additional_gain[units == 'uV'] = 1.
+        self.additional_gain =self.additional_gain.reshape(-1, 1)
+
+    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
+        # in neo rawio channel can acces by names/ids/indexes
+        # there is no garranty that ids/names are unique on some formats
+        raw_traces = self.neo_reader.get_analogsignal_chunk(block_index=self.clock_index, seg_index=self.seg_index,
+                                i_start=start_frame, i_stop=end_frame,
+                               channel_indexes=None, channel_names=None, channel_ids=channel_ids)
+        
+        # rescale gtrace to natural units
+        scaled_traces = self.neo_reader.rescale_signal_raw_to_float(raw_traces, dtype='float32',
+                                    channel_indexes=None, channel_names=None, channel_ids=channel_ids)
```

Comment:
If the data is in `int16`, I'd rather pass the int values and the channel-wise conversion factors that convert and pass `float`s. The reason is that if we use this to read e.g. Plexon data and write it to NWB, we'd be increasing the size of the data when we do not need to.

Reply:
Good idea but at the moemnt there is not official way of handling gain and offset at spikeextractor level.

Reply:
what about making an IntRecordingExtractor? similar to the BinDat one but it would allow to optionally lazily apply offset and gain, and require those at initialization.

Reply:
Honestly I would prefer than the mechanisim of int and rescale to float would be builtin in recordingextarctor not just a special case.